chumari-git/ClipEasy
# Add feature to download YouTube videos

### Summary

This pull request adds a new feature to download YouTube videos using a link in the Django project. The feature includes:

1. A form to accept the YouTube link.
2. A view to handle the form submission and download the video using `pytube`.
3. Templates for the form and success message.
4. Integration with HTML, CSS, and JavaScript frontend.

### Changes

1. Created a new form `YouTubeLinkForm` in `video_downloader/forms.py`.
2. Added a view `download_video` in `video_downloader/views.py` to handle the form submission and download the video.
3. Configured URLs in `video_downloader/urls.py` and included them in the main project `urls.py`.
4. Created templates in `video_downloader/templates/video_downloader/download.html`.
5. Added static files for CSS and JavaScript in the `static` directory.
6. Updated `settings.py` to include `video_downloader` and `django_extensions` in `INSTALLED_APPS`.

### How to Test

1. Navigate to `http://127.0.0.1:8000/video_downloader/download/`.
2. Enter a YouTube link in the form and submit.
3. Verify that the video is downloaded to the `downloads` directory.

### Notes

- Ensure `pytube` is installed in your virtual environment.
- Ensure `django-extensions` is installed and added to `INSTALLED_APPS`.
